File: C# for VB/4-csharp-visual-basic-dotnet-developers-m4-exercise-files/demos/cs/CSModule4/Calculator.cs
using System;

namespace CSModule4
{
    public class Calculator
    {
        public delegate decimal CalculateHandler(decimal x, decimal y);

        //public void Calculate(CalculateHandler operation, decimal x, decimal y)
        //{
        //    decimal result = operation(x, y);
        //    Console.WriteLine("Result: {0}", result);
        //}

        public void Calculate(Action<decimal, decimal> operation, decimal x, decimal y)
        {
            operation(x, y);
            Console.WriteLine("Calculation is complete");
        }

        public void Calculate(Func<decimal, decimal, decimal> operation, decimal x, decimal y)
        {
            decimal result = operation(x, y);
            Console.WriteLine("Result: {0}", result);
        }
    }
}

File: C# for VB/4-csharp-visual-basic-dotnet-developers-m4-exercise-files/demos/vb/VBModule4/Calculator.vb
Public Class Calculator
    Delegate Function CalculateDelegate(x As Decimal, y As Decimal) As Decimal

    'Public Sub Calculate(operation As CalculateDelegate, x As Decimal, y As Decimal)
    '    Dim result = operation(x, y)
    '    Console.WriteLine("Result: {0}", result)
    'End Sub

    Public Sub Calculate(operation As Action(Of Decimal, Decimal), x As Decimal, y As Decimal)
        operation(x, y)
        Console.WriteLine("Calculation complete.")
    End Sub

    Public Sub Calculate(operation As Func(Of Decimal, Decimal, Decimal), x As Decimal, y As Decimal)
        Dim result = operation(x, y)
        Console.WriteLine("Result: {0}", result)
    End Sub

End Class